### PR TITLE
refactor: make peer activePairs private

### PR DESCRIPTION
**Pull request by sangaman**
This commit adds an `activatePair` public method for the `Peer` class and makes the `activePairs` property private. `activatePair` will both add a pair to the set of active pairs as well as request existing orders for that pair from the peer.

### CODE REVIEW

File: lib/orderbook/OrderBook.ts [hunk 1]
```diff
@@ -745,15 +742,9 @@ class OrderBook extends EventEmitter {
     pairIds.forEach(async (pairId) => {
       const [baseCurrency, quoteCurrency] = pairId.split('/');
       if (peer.verifiedCurrencies.has(baseCurrency) && peer.verifiedCurrencies.has(quoteCurrency)) {
-        peer.activePairs.add(pairId);
-        verifiedPairs.push(pairId);
+        await peer.activatePair(pairId);
```
Comment (erkarl) [inline lib/orderbook/OrderBook.ts]: Minor nit: we could do a `pairsToActivate.push(pairId)` -> `await Promise.all(pairsToActivate)` here for slightly better performance in exchange for slightly worse readability. Up to you - I don't mind the way it currently is.

File: test/simulation/tests.go [hunk 2]
```diff
@@ -189,7 +189,7 @@ func testOrderMatchingAndSwap(net *xudtest.NetworkHarness, ht *harnessTest) {
 		Price:    0.02,
 		Quantity: 1000000,
 		PairId:   "LTC/BTC",
-		Side:     xudrpc.OrderSide_BUY,
+		Side:     xudrpc.OrderSide_SELL,
```
Comment (erkarl) [inline test/simulation/tests.go]: Why is this change required?

Comment (sangaman) [reply test/simulation/tests.go]: That's an unintentional leftover, thanks for catching.

File: lib/orderbook/OrderBook.ts (hunk 1 above)
Comment (sangaman) [reply lib/orderbook/OrderBook.ts]: I changed it to use `Promise.all` - not a big deal either way I agree but I like sticking to a pattern of awaiting promises in parallel when they don't depend on each other.
